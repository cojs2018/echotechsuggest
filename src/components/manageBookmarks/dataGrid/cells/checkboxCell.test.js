import React from 'react';
import render from 'react-test-renderer';
import { mount } from 'enzyme';

import CheckboxCell from "./checkboxCell";

describe('CheckboxCell', () => {
    it('renders correctly', () => {
        const props = {
            rowId: 0,
            onChecked: jest.fn()
        }

        const tree = render
            .create(<CheckboxCell {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('handles actions when checked', async () => {
        const props = {
            rowId: 0,
            onChecked: jest.fn(rowId => console.log(`Row ${rowId} has been checked`)),
        }

        let checked = false;
        const setChecked = jest.fn(async (bool) => checked = bool);

        React.useState = jest.fn()
            .mockReturnValueOnce([checked, setChecked]);

        const wrapper = mount(<CheckboxCell {...props} />);

        const checkbox = wrapper.find('Checkbox').first();

        checkbox.props().onPress();
        wrapper.update();

        await new Promise(resolve => setImmediate(resolve));
        expect(checked).toBeTruthy();
    })
});