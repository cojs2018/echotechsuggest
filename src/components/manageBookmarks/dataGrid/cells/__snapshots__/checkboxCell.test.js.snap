// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxCell renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
        undefined,
      ],
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "padding": 6,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "color": "#03dac4",
                "fontSize": 24,
              },
              Array [
                Object {
                  "lineHeight": 24,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ó°„¬
        </Text>
      </View>
    </View>
  </Text>
</View>
`;
